ledger-api/grpc-definitions/com/daml/ledger/api/v2/state_service.proto

```diff
@@ -0,0 +1,140 @@
+  // The format of this field is described in ``ledger_offset.proto``.
+  string offset = 1;
+
+  // For each contract there could be multiple ContractStateComponent-s in the entire snapshot. These together define
+  // the state of the contract in the snapshot.
+  repeated ContractStateComponent contract_state_components = 2;
+}
+
+message ContractStateComponent {
+  oneof contract_state_component {
+    // The contract is active in the given domain.
+    // There will be one such message for each domain the contract is active in.
+    ActiveContract active_contract = 1;
+
+    // This message is only available, if the corresponding assigned event cannot be found before or at the
+    // active_at_offset, and at least one of the querying parties could submit the corresponding assign command.
+    // The incomplete_unassigned.unassigned_event.reassigning_parties will contain all such querying parties.
+    IncompleteUnassigned incomplete_unassigned = 2;
+
+    // Important: this message is not indicating that the contract is active for the target domain!
+    // This message is only available, if the corresponding unassigned event cannot be found before or at the
+    // active_at_offset, and at least one of the querying parties could have submitted the corresponding assign command.
+    // The incomplete_assigned.assigned_event.reassigning_parties will contain all such querying parties.
+    IncompleteAssigned incomplete_assigned = 3;
+  }
+}
+
+message ActiveContract {
+  // Required
+  com.daml.ledger.api.v1.CreatedEvent created_event = 1;
+
+  // A valid domain ID
+  // Required
+  string domain_id = 2;
+
+  // Each corresponding assigned and unassigned event has the same reassignment_counter. This monotonically increases
+  // with each unassign command for the same contract. Creation of the contract corresponds to reassignment_counter
+  // equals zero.
+  // Required
+  uint64 reassignment_counter = 3;
+}
+
+message IncompleteUnassigned {
+  // Required
+  com.daml.ledger.api.v1.CreatedEvent created_event = 1;
+
+  // Required
+  UnassignedEvent unassigned_event = 2;
+}
+
+message IncompleteAssigned {
+  // Required
+  AssignedEvent assigned_event = 1;
+}
+
+
+message GetConnectedDomainsRequest {
+  // The submitting party
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string party = 1;
+}
+
+message GetConnectedDomainsResponse {
+  message ConnectedDomain {
+    // The alias of the domain
+    // Required
+    string domain_alias = 1;
+    // The ID of the domain
+    // Required
+    string domain_id = 2;
+    // The permission on the domain
+    // Required
+    ParticipantPermission permission = 3;
+  }
+  repeated ConnectedDomain connected_domains = 1;
+}
+
+// enum indicating the permission level
+enum ParticipantPermission {
+  MissingParticipantPermission = 0;
+  // participant is active, can submit transactions
+  Submission = 1;
+  // participant is passive, can only confirm transactions
```

Comment:
I'd argue that confirmation is not really passive. Wondering whether we want to drop active/passive here to be sure that it does not create confusion with HA. WDYT?

Reply:
I am not sure, but this might be needed to change together with the source, no?